tasks/main.yml

```diff
@@ -117,32 +123,43 @@
     - ansible_distribution in ['CentOS', 'RedHat']
     - ansible_distribution_version is version('9', '>=')
 
-- name: Join to a specific Domain Controller
-  shell: |
-    set -euo pipefail
-    echo {{ ad_integration_password | quote }} | realm join -U \
-      {{ ad_integration_user | quote }} --membership-software \
-      {{ ad_integration_membership_software | quote }} \
-      {{ ad_integration_join_to_dc | quote }}
+- name: Build Command - Join to a specific Domain Controller
+  set_fact:
+    join_command: |
+      set -euo pipefail
+      echo {{ ad_integration_password | quote }} | realm join -U \
+        {{ ad_integration_user | quote }} --membership-software \
+        {{ ad_integration_membership_software | quote }} \
+        {{ ad_integration_join_parameters }} \
+        {{ ad_integration_join_to_dc | quote }}
   when:
     - ad_integration_join_to_dc is not none
-    - not ansible_check_mode
-  register: __realm_join_output
-  failed_when:
-    - __realm_join_output is failed
-    - not __realm_join_output.stderr is search("Already joined to this domain")
-  changed_when: not __realm_join_output.stderr is
-    search("Already joined to this domain")
 
-- name: Perform discovery-based realm join operation
-  shell: |
-    set -euo pipefail
-    echo {{ ad_integration_password | quote }} | realm join -U \
-      {{ ad_integration_user | quote }} --membership-software \
-      {{ ad_integration_membership_software | quote }} \
-      {{ ad_integration_realm | quote }}
+- name: Build Join Command - Perform discovery-based realm join operation
+  set_fact:
+    join_command: |
+      set -euo pipefail
+      echo {{ ad_integration_password | quote }} | realm join -U \
+        {{ ad_integration_user | quote }} --membership-software \
+        {{ ad_integration_membership_software | quote }} \
+        {{ ad_integration_join_parameters }} \
+        {{ ad_integration_realm | quote }}
   when:
     - ad_integration_join_to_dc is none
+
+- name: Show the join command for debug
+  debug:
+    msg: "Would run: '{{ join_command }}'"
+  when:
+    - ad_integration_join_to_dc == __ad_integration_dummy_dc
+      or ad_integration_realm == __ad_integration_dummy_realm
+      or ansible_check_mode
+
+- name: Run realm join command
+  shell: "{{ join_command }}"  # noqa 305
```

Comment:
add `no_log: true` here - otherwise, the clear text password will be printed in the log

Reply:
Done.